src/abi.rs

```diff
@@ -151,30 +156,41 @@ impl<'gcc, 'tcx> FnAbiGccExt<'gcc, 'tcx> for FnAbi<'tcx, Ty<'tcx>> {
                         argument_tys.push(Reg::i32().gcc_type(cx));
                     }
                     let ty = cast.gcc_type(cx);
-                    apply_attrs(ty, &cast.attrs)
+                    apply_attrs(ty, &cast.attrs, argument_tys.len())
```

Comment:
Why do you pass the length, here?

Reply:
Because we fill `argument_tys` with each argument. So the length is the index of the argument. I'm using the same logic in all `apply_attrs` calls.